### PR TITLE
Add PUBREL for QOS 2 support

### PR DESCRIPTION
**Pull request by rwestergren**
MQTT client QOS 2 flow requires:

```
QOS 2 - exactly one delivery

Normal Flow

client sends a QoS 2 PUBLISH message and stores in Persistent Outbox
broker replies with PUBREC
client sends PUBREL
broker responds with PUBCOMP
client receives PUBCOMP and deletes message from Persistent Outbox
```

### CODE REVIEW

File: phpMQTT.php [hunk 1]
```diff
@@ -259,6 +259,24 @@ function publish($topic, $content, $qos = 0, $retain = 0){
 		fwrite($this->socket, $head, strlen($head));
 		fwrite($this->socket, $buffer, $i);
 
+		if($qos == 2)
```
Comment (optonc) [inline phpMQTT.php]: Why does it not work for me? I am using a emqtt broker.

**Comment by optonc** [comment]
Why it doesn't work for me? I am using a emqtt broker. I wish to send command (publish) by Qos = 2.

**Comment by rwestergren** [comment]
This code tests fine using a [demo broker](http://www.hivemq.com/demos/websocket-client/). For example:
````
// Test Broker
$address = "broker.mqttdashboard.com";
$port = 1883;
$clientid = uniqid();

$mqtt = new phpMQTT($address, $port, $clientid);

$mqtt->connect();
$mqtt->publish("testtopic/1", "Test 123", 2);
$mqtt->close();
````

You may need to troubleshoot a raw packet capture - the issue is likely with your broker.

**Comment by bluerhinos** [comment]
Sorry its taken so long to look at this, but I am back using MQTT now.

I am not going to merge this as its not really supporting QoS 2 as its not waiting for a `PUBREC` before sending the `PUBREL` but I will look into how I can have an 'Outbox' in phpMQTT to support QoS

**Comment by rwestergren** [comment]
Great point - this just fit a specific need for a project I was working on. Closed!

**Pull request closed without merging**